### PR TITLE
*: improve chaos-mesh actions

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What problem does this PR solve? 

improve chaos-mesh tests with GitHub actions.

### What is changed and how it works?

- enable `pod-kill` action with matrix
- add cron schedule to run chaos tests nighlty
- check failed case faster
- add rand `ALTER TABLE` statements
- add network and IO chaos actions
- add shard DDL test cases (only pessimistic now)

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update the `dm/dm-ansible`
 - Need to be included in the release note

### CODE REVIEW

File: chaos/cases/task.go [hunk 1]
```diff
@@ -0,0 +1,395 @@
+// Copyright 2020 PingCAP, Inc.
```
Comment (csuzhangxc) [inline chaos/cases/task.go:1]: this file copied and refactored from `single.go`.

File: chaos/cases/cases.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	"golang.org/x/sync/errgroup"
+
+	config2 "github.com/pingcap/dm/dm/config"
+	"github.com/pingcap/dm/dm/pb"
+	"github.com/pingcap/dm/pkg/utils"
+)
+
+var (
+	// NOTE: items in `doSchemas` should be specified in the corresponding task files (`filenames`).
+	// NOTE: can not support optimistic shard DDL now because go-sqlsmith will generated some statements like
```
Comment (lance6716) [inline chaos/cases/cases.go:29]: how about change it to `TODO` to remind us in future

File: chaos/cases/db.go [hunk 3]
```diff
@@ -122,6 +122,8 @@ func ignoreExecSQLError(err error) bool {
 		return true
 	case errno.ErrDupEntry: // HACK: we tolerate `invalid connection`, then `Duplicate entry` may be reported.
 		return true
+	case errno.ErrTooBigRowsize: // HACK:we tolerate `Error 1118: Row size too large. The maximum row size for the used table type, not counting BLOBs, is 65535`
```
Comment (lance6716) [inline chaos/cases/db.go:125]:
```suggestion
	case errno.ErrTooBigRowsize: // HACK: we tolerate `Error 1118: Row size too large. The maximum row size for the used table type, not counting BLOBs, is 65535`
```

Comment (lance6716) [reply chaos/cases/db.go:125]: is this a connection to MySQL? seems TiDB doesn't have such small length limit

https://docs.pingcap.com/zh/tidb/stable/tidb-limitations

File: Makefile [hunk 4]
```diff
@@ -194,8 +194,8 @@ coverage_fix_cover_mode:
 	sed -i "s/mode: count/mode: atomic/g" $(TEST_DIR)/cov.*.dmctl.*.out
 
 coverage: coverage_fix_cover_mode retool_setup
-	retool do gocovmerge "$(TEST_DIR)"/cov.* | grep -vE ".*.pb.go|.*.__failpoint_binding__.go|.*debug-tools.*|.*portal.*" > "$(TEST_DIR)/all_cov.out"
-	retool do gocovmerge "$(TEST_DIR)"/cov.unit_test*.out | grep -vE ".*.pb.go|.*.__failpoint_binding__.go|.*debug-tools.*|.*portal.*" > $(TEST_DIR)/unit_test.out
+	retool do gocovmerge "$(TEST_DIR)"/cov.* | grep -vE ".*.pb.go|.*.__failpoint_binding__.go|.*debug-tools.*|.*portal.*|.*chaos.*" > "$(TEST_DIR)/all_cov.out"
+	retool do gocovmerge "$(TEST_DIR)"/cov.unit_test*.out | grep -vE ".*.pb.go|.*.__failpoint_binding__.go|.*debug-tools.*|.*portal.*|.*chaos.*" > $(TEST_DIR)/unit_test.out
```
Comment (GMHDBJD) [inline Makefile:198]: Do we need calculate chaos coverage in unit test?

**Review by GMHDBJD — COMMENTED** [review]
rest LGTM

File: chaos/cases/cases.go (hunk 2 above)
Comment (csuzhangxc) [reply chaos/cases/cases.go]: changed in https://github.com/pingcap/dm/pull/1114/commits/91e5cf704830dbd3fc88d361a660f6b958f43b4d

File: chaos/cases/db.go (hunk 3 above)
Comment (csuzhangxc) [reply chaos/cases/db.go]: applied the suggestion in https://github.com/pingcap/dm/pull/1114/commits/91e5cf704830dbd3fc88d361a660f6b958f43b4d.

> is this a connection to MySQL?

YES

> seems TiDB doesn't have such small length limit

TiDB limited a row to 6MB (https://docs.pingcap.com/zh/tidb/stable/tidb-limitations#%E5%8D%95%E8%A1%8C%E7%9A%84%E9%99%90%E5%88%B6)

File: Makefile (hunk 4 above)
Comment (csuzhangxc) [reply Makefile:198]: I don't want to add another UT/IT for these testing codes, so I exclude them for coverage.

File: chaos/scripts/check-case.sh [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    kubectl wait --for=condition=complete job/chaos-test-case --timeout=1m
+    if [ $? -eq 0 ]; then
+        completed=true
+        echo "chaos-test-case has completed"
+        break
+    else
+        echo "chaos-test-case has not completed" ${i}
+        kubectl get job chaos-test-case
+        if [ $? -ne 0 ]; then
+            echo "chaos-test-case job has been cleared"
```
Comment (lance6716) [inline chaos/scripts/check-case.sh:14]: what's the cause of "cleared" 🤔

File: chaos/manifests/network-partition-dm.yaml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  labels:
+    app: network-partition-dm-master-master
+spec:
+  action: partition
+  mode: one
+  selector:
+    labelSelectors:
+      "app": "dm-master"
+  direction: both
+  target:
```
Comment (lance6716) [inline chaos/manifests/network-partition-dm.yaml:40]: does this selector (`target` same as above `seletor`) match all dm-master, so each master is isolated from another?

**Review by lance6716 — COMMENTED** [review]
rest LGTM

File: chaos/scripts/check-case.sh (hunk 5 above)
Comment (csuzhangxc) [reply chaos/scripts/check-case.sh:14]: If the test case job failed, K8s may clear the pod (and also the job) **sometimes** (I don't know how to disable this behavior yet, 😢 )

File: chaos/manifests/network-partition-dm.yaml (hunk 6 above)
Comment (csuzhangxc) [reply chaos/manifests/network-partition-dm.yaml:40]: > does this selector (target same as above seletor) match all dm-master

YES. but we use `mode: one` here, so only one DM-master member will be isolated with another one (leader or follow, maybe we can change to `mode: all` to isolate the selected one with all other members?).

Comment (lance6716) [reply chaos/manifests/network-partition-dm.yaml:40]: Ah we have 3 master now. ~maybe we could use `fixed-percent` or `random-max-percent`~

I think either `one` or `all` is fine, they will test different scenario 😂  what's your opinion @GMHDBJD

**Comment by lance6716** [comment]
LGTM

File: chaos/manifests/network-partition-dm.yaml (hunk 6 above)
Comment (GMHDBJD) [reply chaos/manifests/network-partition-dm.yaml:40]: > I think either one or all is fine, they will test different scenario.
Agree. Maybe add more cases later.

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**